go/private/go_tool_binary.bzl

```diff
@@ -11,14 +11,12 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-load('//go/private:go_toolchain.bzl', 'toolchain_type', 'ConstraintValueInfo', 'go_toolchain_core_attrs')
-
-go_bootstrap_toolchain_type = toolchain_type()
+load('//go/private:go_toolchain.bzl', 'go_toolchain_core_attrs')
 
 def _go_bootstrap_toolchain_impl(ctx):
-  return [go_bootstrap_toolchain_type(
-      exec_compatible_with = ctx.attr.exec_compatible_with,
-      target_compatible_with = ctx.attr.target_compatible_with,
+  return [platform_common.ToolchainInfo(
+      type = Label("@io_bazel_rules_go//go:bootstrap_toolchain"),
```

Comment:
Commit https://github.com/bazelbuild/bazel/commit/bb87c12dda25ebad124907625a9cdb973cf9e1f2 isn't in Bazel 0.5.4, but in the next release this line can be removed. Is it worth trying to get that cherry-picked to 0.5.4 to simplify removing this?